crates/solver/src/settlement.rs

```diff
@@ -218,7 +218,8 @@ impl Interaction for NoopInteraction {
 #[derive(Debug, Clone, Default)]
 pub struct Settlement {
     pub encoder: SettlementEncoder,
-    pub submitter: SubmissionPreference,
+    pub submitter: SubmissionPreference, // todo - extract submitter and score into a separate struct
+    pub score: Option<BigRational>,
```

Comment:
I'm not sure I fully get the purpose of the `score` yet. Is this intended to be a promise that an objective value greater or equal to the `score` will be achieved with this solution?
Wouldn't this only make sense together with the colocated driver which supports RFQ? (RFQ would be the reason why a solver has to promise some result instead of being able to prove it by responding with the entire call data of the solution upfront).
If that is the case wouldn't we have to have a new `Settlement` type which also supports RFQ where the `score` field could be added?

Reply:
The new field `score` added in this PR corresponds to the `reference_score` mentioned in the [CIP17](https://forum.cow.fi/t/cip-draft-auction-model-for-solver-rewards/1405) and it means how much of the reward solver is willing to give up in order to win the auction and right to submit a solution onchain.

Not sure about the rest of the question but my instinct says this fits well into the colocated driver and eliminating simulations and doing proofs afterwards (meaning it is a step towards it). Right now, all changes are related to changing the logic that happens before submission.

Anyhow, since this is a big change itself, I would prefer to only do the changes needed exclusively for solver rewards change, and do the refactoring in a follow up.